mkosi/__init__.py

```diff
@@ -6414,10 +6414,11 @@ def load_args(args: argparse.Namespace) -> MkosiConfig:
             args.mirror = "http://archive.ubuntu.com/ubuntu"
             if platform.machine() == "aarch64":
                 args.mirror = "http://ports.ubuntu.com/"
-        elif args.distribution == Distribution.arch and platform.machine() == "aarch64":
-            args.mirror = "http://mirror.archlinuxarm.org"
-        elif args.distribution == Distribution.arch and platform.machine() == "x86_64":
-            args.mirror = "https://geo.mirror.pkgbuild.com"
+        elif args.distribution == Distribution.arch:
+            if platform.machine() == "aarch64" or args.architecture == "aarch64":
```

Comment:
This is not correct, `args.architecture` will default to "aarch64" if running on a aarch64 machine. We don't have to check explicitly against it here. Please remove the `platform.machine()` check. In other words, this should be the following:

```
if args.architecture == "aarch64"
    args.mirror = "http://mirror.archlinuxarm.org"
else:
    args.mirror = "https://geo.mirror.pkgbuild.com"
```